### PR TITLE
DDCE-5183 - Play 3.0 upgrade

### PR DESCRIPTION
**Pull request by frikit**

### CODE REVIEW

**Review by EdCope — CHANGES_REQUESTED** [review]
Scala 2.13.13 change has made some of the views with boolean params generate warnings if you want to add

File: build.sbt [hunk 1]
```diff
@@ -2,27 +2,21 @@ import play.sbt.routes.RoutesKeys
...
-import uk.gov.hmrc.versioning.SbtGitVersioning.autoImport.majorVersion
 
 lazy val appName: String = "maintain-beneficiaries-frontend"
 
-lazy val root = (project in file("."))
+ThisBuild / scalaVersion := "2.13.13"
+ThisBuild / majorVersion := 0
+
+lazy val microservice = (project in file("."))
   .enablePlugins(PlayScala, SbtDistributablesPlugin, SbtSassify)
```
[the remark this answers is not in the document]
Comment (frikit) [reply build.sbt:12]: fixed

File: build.sbt [hunk 2]
```diff
@@ -2,27 +2,21 @@ import play.sbt.routes.RoutesKeys
...
-    libraryDependencySchemes ++= Seq("org.scala-lang.modules" %% "scala-xml" % VersionScheme.Always),
     scalacOptions ++= Seq(
       "-feature",
       "-Wconf:src=routes/.*:s",
       "-Wconf:cat=unused-imports&src=views/.*:s"
     ),
-    inConfig(Test)(testSettings),
-    majorVersion := 0,
     name := appName,
     RoutesKeys.routesImport += "models._",
```
[the remark this answers is not in the document]
Comment (frikit) [reply build.sbt:21]: fixed

File: project/AppDependencies.scala [hunk 3]
```diff
@@ -2,29 +2,24 @@ import sbt.*
...
-    "uk.gov.hmrc"       %% "play-frontend-hmrc-play-28"     % "8.5.0",
-    "uk.gov.hmrc"       %% "play-conditional-form-mapping"  % "1.13.0-play-28",
-    "uk.gov.hmrc"       %% "domain"                         % "8.3.0-play-28",
-    "org.typelevel"     %% "cats-core"                      % "2.10.0"
+    "uk.gov.hmrc.mongo" %% "hmrc-mongo-play-30"                     % mongoVersion,
+    "uk.gov.hmrc"       %% "bootstrap-frontend-play-30"             % bootstrapVersion,
+    "uk.gov.hmrc"       %% "play-frontend-hmrc-play-30"             % "8.5.0",
+    "uk.gov.hmrc"       %% "play-conditional-form-mapping-play-30"  % "2.0.0",
+    "uk.gov.hmrc"       %% "domain-play-30"                         % "9.0.0",
+    "org.typelevel"     %% "cats-core"                              % "2.10.0"
```
[the remark this answers is not in the document]
Comment (frikit) [reply project/AppDependencies.scala:14]: fixed

File: build.sbt [hunk 4]
```diff
@@ -36,15 +30,11 @@ lazy val root = (project in file("."))
       "controllers.routes._"
     ),
     PlayKeys.playDefaultPort := 9793,
-    ScoverageKeys.coverageExcludedFiles := "<empty>;Reverse.*;.*filters.*;.*handlers.*;.*components.*;" +
-      ".*BuildInfo.*;.*javascript.*;.*FrontendAuditConnector.*;.*Routes.*;.*GuiceInjector;" +
-      ".*ControllerConfiguration;.*LanguageSwitchController",
+    ScoverageKeys.coverageExcludedFiles := "<empty>;.*components.*;.*Mode.*;.*Routes.*;",
```
[the remark this answers is not in the document]
Comment (frikit) [reply build.sbt:33]: will keep it for consistency as it is like this in other repos

[the remark this answers is not in the document]
Comment (EdCope) [reply build.sbt:33]: Ah ok yeh I think is only needed for ExcludedPackages but no problem

**Review by EdCope — APPROVED** [review]

**Pull request merged**